File: BaekJoon/solved/C++/2960 에라토스테네스의 체.cpp
#include <cstdio>
using namespace std;

void BOJ_2960(){
	bool L[1001] = {0,};
    int C = 1, i, j, N, K;
	scanf("%d %d", &N, &K);
    for (i = 2; i <= N; i++)
        for (j = 1; i * j <= N; j++) {
            if (!L[i * j]) {
                if (C == K) {
                    printf("%d", i * j);
                    return;
                }
                L[i * j] = 1;
                C++;
            }
        }
}

int main(void)
{
	BOJ_2960();
}

File: BaekJoon/solved/JS/2960 에라토스테네스의 체.js
let I = (require('fs').readFileSync(0, 'utf-8') + '').trim().split(" ").map(e => +e);

const BOJ_2960 = () => {
    let L = new Array(I[0] + 1).fill(1);
    let C = 1, i, j;
    for (i = 2; i <= I[0]; i++)
        for (j = 1; i * j <= I[0]; j++) {
            if (L[i * j]) {
                if (C == I[1]) {
                    console.log(i * j);
                    return;
                }
                L[i * j] = 0;
                C++;
            }
        }
}

BOJ_2960();